From f15f28c30b67b62cb2aa52b48f2c5c645cbfa170 Mon Sep 17 00:00:00 2001
From: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 26 Jan 2018 15:36:32 +0000
Subject: [PATCH] 	PR target/81763 	* config/i386/i386.md
 (*andndi3_doubleword): Add earlyclobber 	to (=&r,r,rm) alternative. Add
 (=r,0,rm) and (=r,r,0) alternatives.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@257096 138bc75d-0d04-0410-961f-82ee72b054a4
https://gcc.gnu.org/PR81763
https://bugs.gentoo.org/649880
---
 gcc/config/i386/i386.md | 8 ++++----
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/gcc/config/i386/i386.md b/gcc/config/i386/i386.md
index 5cd3ec093cd..fe9649d8738 100644
--- a/gcc/config/i386/i386.md
+++ b/gcc/config/i386/i386.md
@@ -9250,14 +9250,14 @@
 })
 
 (define_insn "*andndi3_doubleword"
-  [(set (match_operand:DI 0 "register_operand" "=r,&r")
+  [(set (match_operand:DI 0 "register_operand" "=&r,r,r,&r")
 	(and:DI
-	  (not:DI (match_operand:DI 1 "register_operand" "r,0"))
-	  (match_operand:DI 2 "nonimmediate_operand" "rm,rm")))
+	  (not:DI (match_operand:DI 1 "register_operand" "r,0,r,0"))
+	  (match_operand:DI 2 "nonimmediate_operand" "rm,rm,0,rm")))
    (clobber (reg:CC FLAGS_REG))]
   "!TARGET_64BIT && TARGET_STV && TARGET_SSE2"
   "#"
-  [(set_attr "isa" "bmi,*")])
+  [(set_attr "isa" "bmi,bmi,bmi,*")])
 
 (define_split
   [(set (match_operand:DI 0 "register_operand")
-- 
2.16.2

